library/src/main/scala/scala/scalajs/runtime/StackTrace.scala

```diff
@@ -153,9 +153,9 @@ object StackTrace {
    *  display the function name.
    */
   private def extractClassMethod(functionName: String): (String, String) = {
-    val PatC = """^ScalaJS\.c\.([^\.]+)(?:\.prototype)?\.([^\.]+)$""".re
-    val PatI = """^(?:Object\.)?ScalaJS\.i\.((?:_[^_]|[^_])+)__([^\.]+)$""".re
-    val PatM = """^(?:Object\.)?ScalaJS\.m\.([^.\.]+)$""".re
+    val PatC = """^(?:ScalaJS\.c\.|\$c_)([^\.]+)(?:\.prototype)?\.([^\.]+)$""".re
+    val PatI = """^(?:Object\.)?(?:ScalaJS\.s\.|\$s_)((?:_[^_]|[^_])+)__([^\.]+)$""".re
+    val PatM = """^(?:Object\.)?(?:ScalaJS\.m\.|\$m_)([^.\.]+)$""".re
```

Comment:
But couldn't we tag the test for the VMs we know it works? I mean the goal is just to test the regexes and that they don't break accidentally because we change the encoding...